### ISSUE TITLE
*inf* value for ppl for "language_modeling"

### ISSUE THREAD

**Issue by nicolabertoldi** [pytorch/fairseq#782]
I am newbie in using fairseq task "*language_modeling" with architecture ransformer_lm.
By using a standard command like this 
```fairseq-train --task language_modeling --arch transformer_lm --save-dir lm_checkpoints/ data_generated
```
where `` contains the data binarised with fairseq-preprocess.
The training data contains about 4M words and have a bpe-tokens dictionary of about 32K.

I see in my log (see below) that ppl gets a *inf* value during the whole training.
Why does this happen?

```
| epoch 043 | loss 2837.172 | ppl inf | wps 22040 | ups 4 | wpb 5116.162 | bsz 4.997 | num_updates 39302 | lr 2.5e-05 | gnorm 17885.347 | clip 1.000 | oom 0.000 | wall 9326 | train_wall 8854
| epoch 043 | valid on 'valid' subset | loss 2322.051 | ppl inf | num_updates 39302 | best_loss 2322.05
| epoch 044 | loss 2819.246 | ppl inf | wps 22042 | ups 4 | wpb 5116.162 | bsz 4.997 | num_updates 40216 | lr 2.5e-05 | gnorm 19552.845 | clip 1.000 | oom 0.000 | wall 9543 | train_wall 9060
| epoch 044 | valid on 'valid' subset | loss 2272.617 | ppl inf | num_updates 40216 | best_loss 2272.62
| epoch 045 | loss 2802.761 | ppl inf | wps 22039 | ups 4 | wpb 5116.162 | bsz 4.997 | num_updates 41130 | lr 2.5e-05 | gnorm 354250.108 | clip 1.000 | oom 0.000 | wall 9761 | train_wall 9266
| epoch 045 | valid on 'valid' subset | loss 2269.807 | ppl inf | num_updates 41130 | best_loss 2269.81
| epoch 046 | loss 2782.943 | ppl inf | wps 22041 | ups 4 | wpb 5116.162 | bsz 4.997 | num_updates 42044 | lr 2.5e-05 | gnorm 30559.840 | clip 1.000 | oom 0.000 | wall 9978 | train_wall 9472
| epoch 046 | valid on 'valid' subset | loss 2250.028 | ppl inf | num_updates 42044 | best_loss 2250.03
| epoch 047 | loss 2769.120 | ppl inf | wps 22042 | ups 4 | wpb 5116.162 | bsz 4.997 | num_updates 42958 | lr 2.5e-05 | gnorm 18006.640 | clip 1.000 | oom 0.000 | wall 10196 | train_wall 9678
| epoch 047 | valid on 'valid' subset | loss 2268.014 | ppl inf | num_updates 42958 | best_loss 2250.03
```

**Comment by lematt1991** [comment]
Perplexity is [computed](https://github.com/pytorch/fairseq/blob/master/fairseq/utils.py#L290) as `2 ^ loss`.  Your `ppl` is `inf` because `2 ^ 2769.120` is very large.  You'll need to modify your hyper-parameters.

**Comment by nicolabertoldi** [comment]
@lematt1991 

thanks to confirm my suspicion
I will try with a better setting of hyper-parameters, and see what happens.
